### PR TITLE
[std] Use $E$ as expression placeholder.

### LINKED ISSUE #3271: Upper-case vs Lower-case for expressions
Throughout the language part of the standard, arbitrary expressions are introduced in lower-case, .e.g:

> The set of _potential results_ of an expression `e` is defined as follows: 

And arbitrary types are introduced in upper-case, e.g.:

> A class type `T` must be complete if:

Both of those are from [basic.def.odr], but it's consistent throughout the language section. 

<hr />

Much of the new library wording from Ranges breaks this convention by introducing expression in upper-case in all the definitions of the CPOs. For instance, from [range.access.begin]:

> The expression `ranges​::​​begin(E)` for some subexpression `E` is expression-equivalent to: 

Is there a particular reason why the Ranges wording uses capital letters for expressions instead of lower-case?

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3271.

### CODE REVIEW

File: source/basic.tex [hunk 1]
```diff
@@ -300,38 +300,38 @@
...
 \grammarterm{id-expression}\iref{expr.prim.id}, the set
-contains only \tcode{e}.
-\item If \tcode{e} is a subscripting operation\iref{expr.sub} with
+contains only $E$.
+\item If $E$ is a subscripting operation\iref{expr.sub} with
 an array operand, the set contains the potential results of that operand.
-\item If \tcode{e} is a class member access
+\item If $E$ is a class member access
 expression\iref{expr.ref} of the form
 \tcode{e1 . \opt{template} e2}
```
Comment (zygoloid) [inline source/basic.tex]: Can we also convert the `e1` and `e2` here and below?

File: source/expressions.tex [hunk 2]
```diff
@@ -600,11 +600,11 @@
...
 When an lvalue-to-rvalue conversion
-is applied to an expression \tcode{e}, and either
+is applied to an expression $E$, and either
 \begin{itemize}
-\item \tcode{e} is not potentially evaluated, or
-\item the evaluation of \tcode{e} results in the evaluation of a member
-      \tcode{ex} of the set of potential results of \tcode{e}, and \tcode{ex}
+\item $E$ is not potentially evaluated, or
+\item the evaluation of $E$ results in the evaluation of a member
+      \tcode{ex} of the set of potential results of $E$, and \tcode{ex}
```
Comment (zygoloid) [inline source/expressions.tex]: `ex` is an expression too.

File: source/basic.tex (hunk 1 above)
Comment (jensmaurer) [reply source/basic.tex]: Fixed.

File: source/expressions.tex (hunk 2 above)
Comment (jensmaurer) [reply source/expressions.tex]: Fixed.

File: source/basic.tex [hunk 3]
```diff
@@ -300,38 +300,38 @@
...
+\item If $E$ is a pointer-to-member
 expression\iref{expr.mptr.oper} of the form
-\tcode{e1 .* e2},
-the set contains the potential results of \tcode{e1}.
-\item If \tcode{e} has the form \tcode{(e1)}, the set contains the
-potential results of \tcode{e1}.
-\item If \tcode{e} is a glvalue conditional
+\tcode{$E_1$ .* $E_2$},
+the set contains the potential results of $E_1$.
+\item If $E$ has the form ($E_1$), the set contains the
```
Comment (zygoloid) [inline source/basic.tex]: We lost the `\tcode` for the parens here.

File: source/expressions.tex [hunk 4]
```diff
@@ -600,13 +600,13 @@
...
 \begin{itemize}
-\item \tcode{e} is not potentially evaluated, or
-\item the evaluation of \tcode{e} results in the evaluation of a member
-      \tcode{ex} of the set of potential results of \tcode{e}, and \tcode{ex}
+\item $E$ is not potentially evaluated, or
+\item the evaluation of $E$ results in the evaluation of a member
+      $E_x$ of the set of potential results of $E$, and $E_x$
       names a variable \tcode{x} that is not odr-used by
-      \tcode{ex}\iref{basic.def.odr},
+      $E_x$\iref{basic.def.odr},
```
Comment (zygoloid) [inline source/expressions.tex]: The subscript x should probably be in code font since that's the same `x` as the variable name.

Comment (jensmaurer) [reply source/expressions.tex]: Fixed.

File: source/basic.tex (hunk 3 above)
Comment (jensmaurer) [reply source/basic.tex]: Fixed.

**Pull request merged**